python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -553,9 +568,6 @@ def add_pending_actor(self, actor: ray.actor.ActorHandle, ready_ref: ray.ObjectR
             actor: The not-yet-ready actor to add as pending to the pool.
             ready_ref: The ready future for the actor.
         """
-        # The caller shouldn't add new actors to the pool after invoking
-        # kill_inactive_actors().
-        assert not self._should_kill_idle_actors
```

Comment:
Is this check necessary? Set to true once start killing the actor, but we still need to create new actors if error occurs.